### DISCUSSION TITLE
Different Target types based on configuration types

### DISCUSSION THREAD

**Discussion by Revbrte** [microsoft/vscode-cmake-tools#2591] [General]
Is it possible to give different target types based on configuration_types?
Foe eg: add_executable when the configuration_type is Debug
             add_library when the configuration type is AB_Debug

**Comment by bobbrow** [comment]
This should be possible with CMake.  I would try something like this (untested):
```
if (CMAKE_BUILD_TYPE MATCHES "Debug")
  add_executable(...)
else()
  add_library(...)
endif()
```

**Reply by bobbrow** [reply]
Note that this probably would not work for multi-config generators because CMAKE_BUILD_TYPE is not set during the configuration step.

**Reply by Revbrte** [reply]
Thanks, I actually tried to configure using command : cmake -H. -B_builds "-DCMAKE_BUILD_TYPE=Debug" -G "Visual Studio 14 2015" -A Win32
and used conditions like this:if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug" )
This works but the problem is when i open the solution file in Visual Studio The target type(exe or dll) of that project for the Debug will be correct. But when i choose  Release/ any other custom config from the Visual Studio drop down it wont be correct.
Is there an alternative to configure correctly for all configuration types using cmake so that when we open the solution file in Visual Studio it will show correct target type for different configuartion types.

**Reply by bobbrow** [reply]
`-G "Visual Studio 14 2015"` is a multi-config generator, so it won't work for that.  You'd have to use `-G "Ninja"` which is a single-config generator.

If you post your question on https://discourse.cmake.org/, you might find someone there who knows how to make it work for multi-config generators.
